src/main/io/ledstrip.c

```diff
@@ -473,6 +473,14 @@ static void applyLedFixedLayers()
 
         case LED_FUNCTION_ARM_STATE:
             color = ARMING_FLAG(ARMED) ? *getSC(LED_SCOLOR_ARMED) : *getSC(LED_SCOLOR_DISARMED);
+            
+            if (ARMING_FLAG(ARMED)) {
+                    nextColor = ledStripConfig()->colors[15];
```

Comment:
I don't like that we are using magic numbers here. This makes configuring the LED_STRIP even more complex.

My hunch is to just disable fading for `LED_FUNCTION_ARM_STATE` (and possibly other functions that have the same problem).

Reply:
I agree,  just liked the trick functionality, seems a shame.  Perhaps the neatest thing to do would be change fade from a modifier to a function in configurator. This way it would be obvious that it is not compatible with the other functions, arm state is the only one that seems to make sense in combination with fade.

I still wouldn't mind the trick numbers though, I feel it would be fine if documented. And who doesn't like obtuse functionality?

Reply:
> And who doesn't like obtuse functionality?

Everybody who has to respond to users opening tickets because they don't understand how to configure stuff... :-|

In that respect, as much as the functionality that you proposed is interesting, the right thing to do is to have colors that are hardcoded into other modifiers **not** be affected by the colour fader.

Reply:
should be simple,  just set previousColor and nextColor to the same value as colour in the function. The fade will still be active, just do nothing as it's fading between the same colours.

Reply:
@MrPlocki: While this would work, it's not ideal in terms of maintainability, since it will not be obvious to somebody looking at the code why the colours are selected in the way they are.

(As a rule of thumb, if your code needs a comment to be understandable, in most cases the solution is to rewrite the code so that it doesn't.)

Reply:
I just tested build 1654 and it seems like the arm state leds are fading from off to their assigned color and back to off. 

I think an aux channel of 1000 should be the default color. I don't see any reason why the default color should be the center. When using a switch or a knob it's easier to find the limit than the center. I also don't think we need so much resolution on shades of the same color. 

What changed from 3.1.7 to 3.2? I thought the behavior was perfect before.

Reply:
There is a wiki entry....... 

It used to just cycle through the hues. Now it looks up the colours set on either side of the selected colour in the palette you see in configurator, it then sets the fade between those three colours. This includes the brightness, where before the LEDs would always stay the same brightness.
This is great if you want to define a custom fade for throttle, and works with an aux channel fade too. However now if it is combined with arm state LEDs, it cannot look up the two colours adjacent as these special colours are set as values and not a palette reference. It then ends up just using whatever is set as the first colour in the palette (default black? ), so try changing that and see what happens. 
We were just discussing the merits of just disabling fade for anything but a static colour. I reluctantly agree that it should be done, but lament the demise of arm state LEDs in combination with quick colour setting via tx aux channel. 

I suppose another option would be to revive the old code and have it do its thing when anything but a solid colour is selected?

Reply:
@Jahnkeanater: https://github.com/betaflight/betaflight/wiki/LED-Strip-functionality describes the new functionality. The aim of it is not to increase resolution, but to increase the flexibility. With the new code, fading can not only be done on hue, but also on brightness and saturation, or a mix of these.